src/ucp/core/ucp_worker.c

```diff
@@ -484,6 +484,54 @@ static ucp_ep_h ucp_worker_find_lane(ucs_list_link_t *ep_list, uct_ep_h uct_ep,
     return NULL;
 }
 
+static unsigned ucp_pending_ff_progress(void *arg)
+{
+    ucs_queue_head_t *tmp_q = arg;
+    unsigned count          = 0;
+    uct_pending_req_t *uct_req;
+
+    ucs_queue_for_each_extract(uct_req, tmp_q, priv, 1) {
+        ucp_ep_err_pending_purge(uct_req, UCS_STATUS_PTR(UCS_ERR_CANCELED));
+        count++;
+    }
+
+    ucs_free(tmp_q);
+    return count;
+}
+
+static ucs_status_t ucp_worker_discard_lane_ff(ucp_worker_h worker,
+                                               uct_ep_h uct_ep)
+{
+    uct_worker_cb_id_t cb_id = UCS_CALLBACKQ_ID_NULL;
+    ucs_queue_head_t *tmp_q  = ucs_malloc(sizeof(*tmp_q),
+                               "discarding tmp queue");
+
+    if (tmp_q == NULL) {
+        ucs_error("cannot allocate memory for discarding tmp queue");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    ucs_queue_head_init(tmp_q);
+    uct_ep_pending_purge(uct_ep, ucp_request_purge_enqueue_cb, tmp_q);
+    if (ucs_queue_is_empty(tmp_q)) {
+        ucs_free(tmp_q);
+        return UCS_OK;
+    }
+
+    /* FLUSH_CANCEL operation might be on pending queue due to
+     * UCS_ERR_NO_RESOURCES, so need to purge the queue to resubmit the
+     * operation. We need to resubmit the FLUSH_CANCEL operation on the same
+     * failed lane, in order to make sure all previous outstanding
+     * operations are completed before destroying the failed endpoint.
+     * @ref ucs_arbiter_t does not support recoursive calls, so schecdule on
```

Comment:
ucp_worker_iface_error_handler is not called from arbiter context so no need to schedule on progress